src/wallet/wallet2.cpp

```diff
@@ -3063,6 +3073,26 @@ bool wallet2::load_keys(const std::string& keys_file_name, const epee::wipeable_
     m_account.set_device(hwdev);
     LOG_PRINT_L0("Device inited...");
   }
+
+  if (r)
+  {
+    if (encrypted_secret_keys)
+    {
+      m_account.decrypt_keys(key);
+    }
+    else
+    {
+      // rewrite with encrypted keys, ignore errors
+      encrypt_keys(key);
```

Comment:
```c++
m_account.encrypt_keys(key);
```

Reply:
I don't think so. We want to leave the keys in "normal" mode here, which is with an encrypted spend key, but unencrypted view key. Since we start with unencrypted keys, encrypt_keys(key); does this, while m_account.encrypt_keys(key); encrypts both. Is that wrong ?

Reply:
You're right.

Reply:
Wait, I think it should check if the encryption is enabled:
```
if (m_ask_password && !m_rpc)
  encrypt_keys(key);
```
Likewise for the subsequent decryption step:
```
if (m_ask_password && !m_rpc)
  decrypt_keys(key);
```
To reproduce this issue: 1) use the release CLI to create a wallet, and `set ask-password 0`, 2) open the wallet with this patch, 3) close it, and 4) try to open the same wallet file again, which fails.